aodndata/soop/soop_ba.py

```diff
@@ -0,0 +1,119 @@
+            - text, doc, xml, images: basic checks
+              uploaded to S3 => set check_type and publish_type attributesge accordingly
+        """
+        if self.file_type is FileType.ZIP:
+            netcdf = self.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)
+            # if uploaded file is ZIP check that ZIP contains a NetCDF
+            if len(netcdf) != 1:
+                raise InvalidInputFileError(
+                    "Expecting one netCDF file in ZIP archive '{zip}'".format(zip=os.path.basename(self.input_file))
+                )
+
+            # first process the NetCDF file to set the destination path for the file collection.
+            nc = netcdf[0]
+            nc.check_type = PipelineFileCheckType.NC_COMPLIANCE_CHECK
+            destination = SoopBaFileClassifier.dest_path(nc)
+            nc.dest_path = os.path.join(destination, nc.name)
+
+            #  delete previous version base on filetype/extension
+            # netcdf file have to be deleted since filename contains creation date=>never overwritten
+            results = self.state_query.query_storage(destination)
+            if results:
+                get_previous_version(self, results, nc.name, destination)
+
+            non_nc_files = PipelineFileCollection(f for f in self.file_collection if f.file_type is not FileType.NETCDF)
+            for non_nc in non_nc_files:
+                non_nc.check_type = PipelineFileCheckType.FORMAT_CHECK
+                if non_nc.extension in ['.ek5', '.out', '.raw']:
+                    non_nc.publish_type = PipelineFilePublishType.ARCHIVE_ONLY
+                    dest_archive = SoopBaFileClassifier.archive_path(nc)
+                    non_nc.archive_path = os.path.join(dest_archive, non_nc.name)
+                else:
+                    non_nc.publish_type = PipelineFilePublishType.UPLOAD_ONLY
+                    non_nc.dest_path = os.path.join(destination, non_nc.name)
+                    if results:
+                        get_previous_version(self, results, non_nc.name)
+
+        else:  # singe netcdf file
+            netcdf = self.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)
+            nc = netcdf[0]
+            nc.check_type = PipelineFileCheckType.NC_COMPLIANCE_CHECK
+            destination = SoopBaFileClassifier.dest_path(nc)
+            nc.dest_path = os.path.join(destination, nc.name)
+
+            #  delete previous version base on filetype/extension
+            # netcdf file have to be deleted since filename contains creation date=>never overwritten
+            results = self.state_query.query_storage(destination)
+            if results:
+                get_previous_version(self, results, nc.name, destination)
+
+
+    dest_path = SoopBaFileClassifier.dest_path
+    archive_path = SoopBaFileClassifier.archive_path
+
+
+def get_previous_version(self, previous_file_list, input_file):
```

Comment:
I don't understand how this works since it's not indented properly to be part of the class. Same with function def below.